Veda Application Chart Changes

Charitha2009 opened the pull request:
# Description
<img width="736" height="545" alt="Screenshot 2026-03-14 at 10 20 30 PM" src="https://github.com/user-attachments/assets/a3df3a6b-00fd-45df-b270-983a12ef43e4" />
<img width="714" height="272" alt="Screenshot 2026-03-14 at 10 20 44 PM" src="https://github.com/user-attachments/assets/10fcab54-00d6-4a8f-a031-f5f2260d2728" />
<img width="716" height="348" alt="Screenshot 2026-03-14 at 10 20 58 PM" src="https://github.com/user-attachments/assets/35d6222a-27d2-42c0-9ecf-d29909a0a345" />
<img width="744" height="491" alt="Screenshot 2026-03-14 at 10 21 11 PM" src="https://github.com/user-attachments/assets/feaa3345-81b4-4f89-b65b-1d5146b91a4f" />

## Related PRS (if any):

Frontend PR: This PR
Backend [PR: #1798 - Country Map Changes](https://github.com/OneCommunityGlobal/HGNRest/pull/1798)

## Main changes explained:

- Created CountryApplicationMap.jsx under /components/Analytics/
- Renders world map using react-simple-maps with the portal’s existing map theme.
- Integrates gradient color scale (light-to-dark) based on the number of applications per country.
- Added hover tooltip using react-tooltip to show:
- Exact applicant count per country
- Percentage change compared to the previous time frame.
- Conditional logic to disable comparison text for custom date ranges.
- Introduced time filter dropdown (Weekly, Monthly, Yearly, Custom)
- Dynamically updates the dataset via API call to fetch filtered analytics.
- Implemented multi-select dropdown for roles using react-select.
- Filters the map view to display data relevant only to selected roles.
- Configured API integration using React Query + Axios for efficient data fetching and caching.
- Updated /pages/JobAnalytics.jsx to include the map component within the Job Posting Analytics dashboard layout.
- Added state management hooks (useState, useEffect, useQuery) for synchronization between date, role, and map data.
- Adjusted responsive design for accessibility across screen sizes and ensured color contrast compliance.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to http://localhost:5173/country-application-map
6. Verify the following functionality:

- Map title is displayed correctly
- Color gradient reflects applicant density
- Hovering over a country shows applicant count and comparison percentage
- Selecting Weekly/Monthly/Yearly updates data dynamically
- Selecting Custom Range disables percentage comparison
- Selecting multiple roles updates map data accurately
- Works properly in dark mode

## Screenshots or videos of changes:
<img width="1214" height="734" alt="Screenshot 2025-10-10 at 1 15 36 AM" src="https://github.com/user-attachments/assets/ec4bb303-5b5e-4d90-bea0-dd9e3b2574bf" />
<img width="216" height="467" alt="Screenshot 2025-10-10 at 1 16 01 AM" src="https://github.com/user-attachments/assets/e9c19901-bc7b-4b33-aa89-cf6c19d3118b" />
<img width="207" height="462" alt="Screenshot 2025-10-10 at 1 16 27 AM" src="https://github.com/user-attachments/assets/9364b28c-a8c7-441e-9b24-659e0100dbfa" />
<img width="1212" height="711" alt="Screenshot 2025-10-10 at 1 16 50 AM" src="https://github.com/user-attachments/assets/b845946e-eea9-4324-bfbe-34d27cc76eeb" />

Commit:
feat: Add Country of Application Map Chart with filtering and tooltips

- Added interactive world map showing application counts by country
- Implemented time period filters (Last 7 Days, Last 30 Days, Last Year, All Time, Custom)
- Added role filter dropdown for filtering by applicant role
- Created custom tooltips showing application count and percentage change
- Used pastel green color scale for country shading
- Added summary statistics (Total Applications, Countries, Avg per Country)
- Integrated with backend API endpoint /analytics/country-applications
- Includes dark mode support
- Added service layer for API calls with mock data fallback
- Accessible at /country-application-map route
- Fixed lint-staged configuration to use npx for stylelint

Commit:
fix: Remove duplicate files and add eslint comment for console.error

ShradhaMBhadrannavar reviewed (changes requested):
The map title, color gradient, and hover details all display correctly. Data updates properly for Weekly, Monthly, Yearly, and Custom Range selections, and everything works well in dark mode.
The only issue is that multiple role selection isn’t possible  - the filter works fine otherwise.
<img width="1470" height="956" alt="Screenshot 2025-10-10 at 20 58 54" src="https://github.com/user-attachments/assets/0bbf4c64-ab1b-4515-b43b-f649f0c6b490" />

harshavarma29 reviewed (changes requested):
I have reviewed the PR locally using the Admin account. The map is responsive and works fine in dark mode. However, I noticed the following issues:

1) The ‘Application Count’ received from the backend does not match the ‘Application Count’ displayed in the map chart. The below image shows one of the example:
    <img width="1906" height="888" alt="data-not-matching" src="https://github.com/user-attachments/assets/0c8ddb28-7646-45ef-9241-522e71bbb525" />

2) The date filter does not allow selecting a complete date (day, month, and year). It closes immediately after selecting the month and changes the state of the application.
    https://github.com/user-attachments/assets/95732dc6-a624-499b-8f36-3cd30d080997

IndraAnuraag9 reviewed (changes requested):
I have tested this PR locally using an Admin account. Everything works properly in both light and dark mode. Country Application Map page is responsive and loads correctly with proper title, color gradient. Hover, range selection, and role filters update data accurately. 
The only issue is that, the multiple role selection isn’t possible.

<img width="1440" height="900" alt="Screenshot 2025-10-24 at 4 43 32 PM" src="https://github.com/user-attachments/assets/b0d50b2a-8dd3-4f7c-a552-02caa30de352" />
<img width="1440" height="900" alt="Screenshot 2025-10-24 at 4 43 40 PM" src="https://github.com/user-attachments/assets/a39e8ce5-8116-4964-b90d-fa25688861dc" />
<img width="1440" height="900" alt="Screenshot 2025-10-24 at 4 43 52 PM" src="https://github.com/user-attachments/assets/e65df8a4-b5c2-4076-8fc5-f8b0ebdc80ea" />

Anusha-Gali reviewed (changes requested):
Hi Veda,

I have reviewed your PR locally and have identified the below issues:
- There no data shown when "All time" and "All roles" is selected
- The calendar allows me to select previous day in to date
- As mentioned by other reviewers, multi-select dropdown for roles using react-select is not available
<img width="1264" height="824" alt="Screenshot 2025-12-10 at 7 46 04 PM" src="https://github.com/user-attachments/assets/05b5cfc4-36ec-4123-8816-ce51c25f27fd" />
<img width="1264" height="824" alt="Screenshot 2025-12-10 at 8 07 24 PM" src="https://github.com/user-attachments/assets/cd4c3690-e2a9-4962-b7c7-118f28c1acc7" />

Commit:
Merge origin/development into veda-country-map-changes

Made-with: Cursor

Commit:
Update yarn.lock after merge

Made-with: Cursor

Commit:
Merge origin/development into veda-country-map-changes

Resolve yarn.lock conflicts for ansi-escapes and type-fest using development structure.

Made-with: Cursor

Commit:
Fix country map filters, date pickers, and dark-mode calendar styling

- Send timeFrame=ALL and rolling start/end dates so API matches UI labels
- Load role options from /api/analytics/roles; token from localStorage or sessionStorage
- Replace native date inputs with react-datepicker; dark calendar popover and white icon
- URL builder sends timeFrame including ALL

Made-with: Cursor

Anusha-Gali reviewed (changes requested):
Hi Veda,

Thank you for working on the issues, however few issues still remain:
- The multi select option is still not available
- There is no comparison text being shown on hover
- In responsive mode, there is huge empty space between the bar metrics and KPI tiles

<img width="1274" height="760" alt="Image" src="https://github.com/user-attachments/assets/cc229a62-e9e1-44ae-bded-cc1a9db46862" />
<img width="1274" height="760" alt="Image" src="https://github.com/user-attachments/assets/3657a6ee-cf7f-4dad-bf7c-9f98877a0aad" />
<img width="1274" height="760" alt="Image" src="https://github.com/user-attachments/assets/8af8df38-ecda-477c-b8a0-5fa892427947" />
<img width="1274" height="760" alt="Image" src="https://github.com/user-attachments/assets/cff3fd9d-f58f-49a0-a9b5-48f0be1cb59e" />
<img width="1274" height="760" alt="Image" src="https://github.com/user-attachments/assets/be423a6f-5d26-4a83-9ada-259cd1b3816d" />
<img width="1274" height="760" alt="Image" src="https://github.com/user-attachments/assets/38ae333d-c39e-4868-b7d5-13162056ed2b" />

Commit:
Merge origin/development into veda-country-map-changes

Resolve yarn.lock conflict: remove duplicate commander@^2.20.0 block (already
covered by commander@2, commander@^2.20.0 entry).

Made-with: Cursor

Commit:
Country map: tooltip presets, colors, multi-select dark mode, deps

- Log-scaled map colors and stronger contrast for application spread
- Tooltip: locale counts, prior period for WEEK/MONTH/YEAR, comparison text tied to preset
- Fix comparison phrase defaulting to last year; clear hover on refetch
- react-select: :global styles and html/body.dark-mode for portaled menu and tooltip
- Ensure react-simple-maps in yarn.lock

Made-with: Cursor

rajanidi1999 reviewed (dismissed):
Hi,
I have successfully reviewed your PR in locally..

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/ac0e439a-8937-432c-a8b9-b8217036bba2" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/4eb9e12c-0853-4615-9c4f-4213fe1b7f6f" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/608769ee-2ec4-440a-be0a-40ce765a7d6e" />

Anusha-Gali reviewed (approved):
Hi Veda,

Thank you for fixing the issues, the functionality works as per requirement.

<img width="1271" height="745" alt="Image" src="https://github.com/user-attachments/assets/7ff08185-8757-41af-9eac-c2d30356defc" />
<img width="1271" height="745" alt="Image" src="https://github.com/user-attachments/assets/bf632dbf-748d-4d60-9320-1bc7ae862ff1" />
<img width="1271" height="745" alt="Image" src="https://github.com/user-attachments/assets/53ab8b82-f968-4bdc-9c58-73e8df46e62c" />
<img width="1271" height="745" alt="Image" src="https://github.com/user-attachments/assets/c02e4f81-7c99-4edc-ad60-1ea40cd8c12b" />

Commit:
Merge origin/development into veda-country-map-changes

Resolve yarn.lock merge conflict by taking development and regenerating lockfile with yarn install.

Made-with: Cursor

Commit:
Fix stylelint blockers after development merge

Apply merge-related CSS lint fixes and targeted rule suppressions required for repository pre-push checks.

Made-with: Cursor

one-community commented:
Thank you all, merging!
